lib/block_view_validator.go

```diff
@@ -1937,6 +1937,69 @@ func (bav *UtxoView) GetGlobalActiveStakeAmountNanos() (*uint256.Int, error) {
 	return globalActiveStakeAmountNanos, nil
 }
 
+func (bav *UtxoView) JailAllInactiveValidators(blockHeight uint64) error {
+	// First, iterate through all of the !isDeleted ValidatorEntries in the UtxoView and
+	// jail any that are inactive.
+	var utxoViewValidatorPKIDs []*PKID
+	for _, validatorEntry := range bav.ValidatorPKIDToValidatorEntry {
+		// We don't want to retrieve any ValidatorEntries from the db that are present in the UtxoView.
+		utxoViewValidatorPKIDs = append(utxoViewValidatorPKIDs, validatorEntry.ValidatorPKID)
+
+		if validatorEntry.isDeleted {
+			continue
+		}
+
+		// Check if we should jail the validator.
+		shouldJailValidator, err := bav.ShouldJailValidator(validatorEntry, blockHeight)
+		if err != nil {
+			return errors.Wrapf(
+				err,
+				"JailAllInactiveValidators: problem determining if should jail validator %v: ",
+				validatorEntry.ValidatorPKID,
+			)
+		}
+
+		// Jail them if so.
+		if !shouldJailValidator {
+			continue
+		}
+
```

Comment:
```suggestion
// If we're here, this validator should be jailed.
```

or something like that.